modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminManagementToolbarDisplayContext.java

```diff
@@ -500,6 +491,52 @@ private String _getTemplatePath() {
 		return _portletConfig.getInitParameter("template-path");
 	}
 
+	private boolean _hasAddKBArticlePermission() throws PortalException {
+		long kbFolderClassNameId = PortalUtil.getClassNameId(
+			KBFolderConstants.getClassName());
+
+		long parentResourceClassNameId = ParamUtil.getLong(
+			_httpServletRequest, "parentResourceClassNameId",
+			kbFolderClassNameId);
+
+		if (parentResourceClassNameId == kbFolderClassNameId) {
+			long parentResourcePrimKey = ParamUtil.getLong(
+				_httpServletRequest, "parentResourcePrimKey",
+				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+			return KBFolderPermission.contains(
+				_themeDisplay.getPermissionChecker(),
+				_themeDisplay.getScopeGroupId(), parentResourcePrimKey,
+				KBActionKeys.ADD_KB_ARTICLE);
+		}
+
+		return AdminPermission.contains(
+			_themeDisplay.getPermissionChecker(),
+			_themeDisplay.getScopeGroupId(), KBActionKeys.ADD_KB_ARTICLE);
+	}
+
+	private boolean _hasAddKBFolderPermission() throws PortalException {
```

Comment:
We are adding 2 times the folder on the creation menu

![image](https://user-images.githubusercontent.com/47524751/177334938-3eb7cc92-ab27-428e-a33b-cf6cb5f3c5ca.png)